src/main/webapp/javascript/login.js

```diff
@@ -0,0 +1,116 @@
+function switchRegistration() {
+    hideError();
+    var checkbox = document.getElementById("register-cb");
+    var checkbox_carRegButton = document.getElementById("register_car-grp");
+    var checkbox_carReg = document.getElementById("register_car-cb");
+    var pwd2 = document.getElementById("password2-grp");
+    var loginBtn = document.getElementById("login-btn");
+    var registerBtn = document.getElementById("register-btn");
+    var name = document.getElementById("name-grp");
+    var surname = document.getElementById("surname-grp");
+    var phoneNumber = document.getElementById("phoneNumber-grp");
+    if (checkbox.checked) {
+        pwd2.classList.remove("w3-hide");
+        name.classList.remove("w3-hide");
+        surname.classList.remove("w3-hide");
+        phoneNumber.classList.remove("w3-hide");
+        loginBtn.classList.add("w3-hide");
+        registerBtn.classList.remove("w3-hide");
+    } else {
+        pwd2.classList.add("w3-hide");
+        loginBtn.classList.remove("w3-hide");
+        registerBtn.classList.add("w3-hide");
+        name.classList.add("w3-hide");
+        surname.classList.add("w3-hide");
+        phoneNumber.classList.add("w3-hide");
+    }
+}
+function login() {
+    hideError();
+    console.log("start login");
+    var usernameTxt = document.getElementById("username-txt");
+    var passwordTxt = document.getElementById("password1-txt");
+    var dto = {
+        "username": usernameTxt.value,
+        "password": passwordTxt.value
+    };
+    console.log("sending login data");
+    fetch('/api/auth/login', {
+        "method": "POST",
+        headers: {
+            'Accept': 'application/json',
+            'Content-Type': 'application/json'
+        },
+        body: JSON.stringify(dto)
+    }).then(function (response) {
+        if (response.status === 200) {
+            console.log("login success");
+            location.href = "/main.jsp";
+        } else {
+            showError("Username or Password is incorrect!");
+        }
+    })
+}
+function register() {
+    hideError();
+    var usernameTxt = document.getElementById("username-txt");
+    var password1Txt = document.getElementById("password1-txt");
+    var password2Txt = document.getElementById("password2-txt");
+    var name = document.getElementById("name-txt");
+    var surname = document.getElementById("surname-txt");
+    var phoneNumber = document.getElementById("phoneNumber-txt");
+    var pwd1 = password1Txt.value;
+    var pwd2 = password2Txt.value;
+    if (pwd1 !== pwd2) {
+        showError("Passwords doesn't match!")
+        return;
+    }
+    var dto = {
+        "username": usernameTxt.value,
+        "password": pwd1,
+        "name": name.value,
+        "surname": surname.value,
+        "phoneNumber": phoneNumber.value
+    };
```

Comment:
The above code, which collects information from UI into a DTO, should be extracted into its own function.